codec-http/src/main/java/io/netty/handler/codec/http/HttpStatusClass.java

```diff
@@ -56,22 +56,20 @@ public boolean contains(int code) {
      * Returns the class of the specified HTTP status code.
      */
     public static HttpStatusClass valueOf(int code) {
-        if (INFORMATIONAL.contains(code)) {
-            return INFORMATIONAL;
+        switch (code / 100) {
```

Comment:
any JMH number? and tableswitch may slower than  hand arranged if else.

Reply:
I think 'tableswitch' is faster than 'if else' in most cases. U may mean the 'lookupswitch'?

Reply:
Depending the number of cases (below 10), the JIT just use a binary-like  cascade of if, regardless the bytecode used

Reply:
@franz1981 The number of cases threshold may depends on the CPU type, for example, the 'MinJumpTableSize' is 5 in sparc: https://github.com/openjdk/jdk8u/blob/master/hotspot/src/cpu/sparc/vm/c2_globals_sparc.hpp#L60.
Further, in common cases, I think the binary-search is faster than the non-optimized 'if-else'.

Reply:
Yeah, sorry, I tried remember by heart it, but I am biased toward my poor old X86 laptop :P
Said that, I believe bothering about the bytecode used isn't the key point here....but it's a fun exercise to microbench it, to be sure. And it requires some perfasm analysis.

> Further, in common cases, I think the binary-search is faster than the non-optimized 'if-else'.

It really depends by the use cases

Reply:
Yes, that's why the `SERVER_ERROR`  is the last.

Reply:
@franz1981 , Appreciate for the benchmarks reference, will check it. As you said, the division may be the real bottleneck. I will run the benchmark to verify it.
@He-Pin , I think put `SERVER_ERROR` the last doesn't mean it will be faster than binary-search, it depends by the use cases.
If the CPU load goes heavy and cause more `SERVER_ERROR` code, then put `SERVER_ERROR` the last will amplify the load and make things going worse.